modules/roles_profiles/manifests/profiles/notarization_users.pp

```diff
@@ -0,0 +1,54 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+class roles_profiles::profiles::notarization_users {
+    case $::operatingsystem {
+        'Darwin': {
+            include sudo
+            # Make sure the users profile is required
+            # That is where the user virtual resources are generated
+            require roles_profiles::profiles::users
+            # Lookup the group array and realize their user resource
+            $users = $::hostname ? {
+                /^(tb-)?mac-v3-signing\d+/ =>
+                    lookup('user_groups.notarization', Array, undef, []),
+                /^dep-mac-v3-signing\d+/ =>
+                    lookup('user_groups.dep_notarization', Array, undef, []),
+                default => [],
+            }
+            realize(Users::Single_user[$users])
+
+            # Monkey patching directoryservice.rb in order to create users also breaks group merging
+            # So we directly add the user to the group(s)
+            $users.each |String $user| {
+                exec { "${user}_developer_group":
+                    command => "/usr/bin/dscl . -append /Groups/admin GroupMembership ${user}",
```

Comment:
Good catch, need to double check with @nthomas

Reply:
Oops, wrong user. @nthomas-mozilla

Reply:
I think the important thing is that notarization users (which are going to talk to apple) have read access to the zip file they need to upload, and the upload tool (`xcrun altool`). It's likely that making them an Admin was the quick way to do this when we were standing this up manually. I don't know what the developer group is, but it might solve the 2nd issue.